src/tests/mod.rs

```diff
@@ -1901,3 +1903,34 @@ macro_rules! generate_ops_tests {
         }
     };
 }
+
+#[test]
+fn fraction_from_float() {
```

Comment:
Thanks for adding the tests for it! Could you also add some tests with `BigFraction` in here?